### PR TITLE
add datetime attribute that works with mypy

### PR DESCRIPTION
**Pull request by knap1930**
probably not the best name, as it may not be clear it will convert to UTC?

### CODE REVIEW

File: pynamodb_attributes/datetime.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from pynamodb.attributes import Attribute
+from pynamodb.attributes import UTCDateTimeAttribute
+
+
+class DateTimeAttribute(Attribute):
+    """
+    Unlike UTCDateTimeAttribute, this attribute has its type hinted as 'datetime'.
```
Comment (ikonst) [inline pynamodb_attributes/datetime.py:7]: https://github.com/pynamodb/PynamoDB/blob/140993f453e7a52d99af5a9835b87aade648e47e/pynamodb/attributes.pyi#L121

How is this different?

File: tests/unicode_delimited_tuple_attribute_test.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def test_serialization_untyped(expected_attributes, value, uuid_key):
 
     # verify underlying storage
     item = MyModel._get_connection().get_item(uuid_key)
-    assert item == {'Item': {'key': ANY, **expected_attributes}}
+    assert item == {'ConsumedCapacity': ANY, 'Item': {'key': ANY, **expected_attributes}}
```
Comment (ikonst) [inline tests/unicode_delimited_tuple_attribute_test.py:135]: Hm, I wonder if it's a new boto version

File: pynamodb_attributes/datetime.py (hunk 1 above)
Comment (zskyfly) [reply pynamodb_attributes/datetime.py:7]: isn't the problem that `UTCDateTimeAttribute` _does_ have its type hinted as `datetime` and so the `set` method fails the mypy check?

Comment (ikonst) [reply pynamodb_attributes/datetime.py:7]: Could you give me an example of code that fails with plain PynamoDB and will be fixed by this?

Comment (knap1930) [reply pynamodb_attributes/datetime.py:7]: This is the code thats failing:
https://github.com/lyft/cancels/blob/d4795270a8a0177d9af27498091a6821a1ed6fb1/app/db/async_blocker.py#L30-L31

This is the mypy error:
```
app/db/async_blocker.py:30: error: "datetime" has no attribute "set"
app/db/async_blocker.py:30: error: Unsupported left operand type for | ("datetime")
app/db/async_blocker.py:31: error: "datetime" has no attribute "set"
```

Comment (ikonst) [reply pynamodb_attributes/datetime.py:7]: Are you adding pynamodb to your mypy dependencies?

(I suspect you're using the outdated PynamoDB stubs that come with the Python typeshed.)

Comment (knap1930) [reply pynamodb_attributes/datetime.py:7]: when adding `pynamodb` to  `additional_dependencies` the error changed to

```
app/db/async_blocker.py:30: error: Unsupported left operand type for | ("UTCDateTimeAttribute")
```

Comment (ikonst) [reply pynamodb_attributes/datetime.py:7]: The culprit is that PynamoDB's `Attribute` type stub doesn't define
```
def __or__() -> pynamodb.expressions.operand._IfNotExists: ...
```
and similarly `_IfNotExists` isn't defined in operand.pyi.

cc @garrettheel

**Comment by knap1930** [comment]
talked to Ilya on slack, will create a PR to solve this in PynamoDB soon.

**Pull request closed without merging**